### PR TITLE
[firmware-upgrader] Set up travis build and qa checks #1

### LINKED ISSUE #1: [firmware-upgrader] Set up travis build and qa checks
Set up a travis build in line with the other openwisp modules (see django-freeradius, openwisp-ipam, for recent good examples).

The build should test the latest versions of django (2.2 and 3.0) and perform all the qa-checks done in the other repos.

### PR DESCRIPTION
**Pull request by IAmAMissingColon**
Configured Travis-CI

Closes #1

### CODE REVIEW

**Comment by IAmAMissingColon** [comment]
Review Needed @nemesisdesign :smile:

**Comment by IAmAMissingColon** [comment]
finding solution on this `error: openwisp-utils 0.3.2 is installed but openwisp-utils<0.3,>=0.2 is required by {'django-netjsonconfig'}`

File: .travis.yml [hunk 1]
```diff
@@ -26,12 +32,15 @@ before_install:
   - ./runisort
 
 install:
+  - pip install $DJANGO
   - python setup.py -q develop
-  - if [[ $TRAVIS_PYTHON_VERSION == "2.7" ]]; then pip install git+git://github.com/tinio/pysqlite.git@extension-enabled#egg=pysqlite; fi
 
 script:
   - coverage run --source=openwisp_firmware_upgrader runtests.py
-  - if [[ $TRAVIS_PYTHON_VERSION == 3.4 ]]; then ./tests/manage.py makemigrations --dry-run | grep "No changes detected"; fi
```
Comment (IAmAMissingColon) [inline .travis.yml]: Do I Need To Keep This ?

Comment (nemesisdesign) [reply .travis.yml]: no because it's done by `openwisp-utils-qa-checks`

Comment (IAmAMissingColon) [reply .travis.yml]: Okay :smiley:

**Comment by IAmAMissingColon** [comment]
Sorry! I Mistakenly Reviewed My PR :/

**Comment by TheOneAboveAllTitan** [comment]
In your setup.py, `openwisp-controller>=0.3.1,<0.4.1` is used installing openwisp-controller==0.4 which uses `django-netjsonconfig==0.9` that installs `openwisp-utils>=0.2,<0.3`. 

I think that is the reason which is making Travis fail. It should be fixed by #6

**Comment by IAmAMissingColon** [comment]
> @IAmAMissingColon you can now rebase on the current master and proceed

Done :relaxed:

**Comment by IAmAMissingColon** [comment]
Like That @nemesisdesign :smiley:

**Pull request merged**